At member creation check for IP conflict in subnet

BenjaminLudwigSAP opened the pull request:
- [x] Add functional tests
- [x] Check at member creation
- [x] Check at loadbalancer (vip) creation

notandy commented on octavia/f5_extensions/workarounds.py:
```diff
@@ -0,0 +1,40 @@
+
+    # get VIPs from network
+    try:
+        conflicts = repositories.VipRepository().get(db_api.get_session(autocommit=False), subnet_id=member_subnet)
```
This just checks if a VIP from the same subnet is used. This will be probably always true and block like every member call.

notandy replied:
Also, please keep in mind that _any_ vip ip_address in the same network should be prohibited, not only from the same subnet. the subnet_id is an optional member property.

laerlingSAP replied:
Those two comments sound paradoxical. Same VIP from the same subnet should be allowed but not from the same network?
I suppose you only mean the latter.

notandy reviewed (changes requested):
Please add Unit test (success and failure condition)

BenjaminLudwigSAP committed:
At member creation check for conflicting VIP in network

BenjaminLudwigSAP committed:
At load balancer creation check for conflicting member IP in network

BenjaminLudwigSAP committed:
Fix VIP/member conflict, add functional tests

f5_workarounds.check_loadbalancer_for_invalid_ip was called in the
wrong location, where the VIP was not completely populated yet.

BenjaminLudwigSAP committed:
test_f5_extensions: Fix copyright notice

notandy committed:
Re-use lock_session and repository instances for ip conflict check

Change-Id: Ib67ab93b401a571c2c4832bfd6d3bea3e4ec1079
